lib/dgram.js

```diff
@@ -631,6 +631,49 @@ Socket.prototype.dropMembership = function(multicastAddress,
   }
 };
 
+Socket.prototype.addSourceSpecificMembership = function(sourceAddress,
+                                                        groupAddress,
+                                                        interfaceAddress) {
+  this._healthCheck();
+
+  if (!sourceAddress) {
```

Comment:
Can you make these argument checks stricter by checking for specific supported data types.

Reply:
I would drop the `if (!sourceAddress)` (and similar checks below) and just leave the check for `typeof === 'string'`. In general, I'd keep away from `if (var)` and `if (!var)` when doing type checking because they can match so many different things.

Reply:
fixed